crates/factor-key-value/src/lib.rs

```diff
@@ -19,22 +20,22 @@ use spin_key_value::{
 };
 pub use store::MakeKeyValueStore;
 
-pub struct KeyValueFactor {
-    runtime_config_resolver: Arc<dyn runtime_config::RuntimeConfigResolver>,
+pub struct KeyValueFactor<C> {
+    runtime_config_resolver: Arc<dyn runtime_config::RuntimeConfigResolver<Config = C>>,
```

Comment:
If this is going to need a generic it seems like it might as well use that for the `RuntimeConfigResolver` instead. Then you could `type RuntimeConfig = RuntimeConfig<T::Config>`.

Reply:
Seems to work out: https://github.com/fermyon/spin/commit/03281141b4446b4fa58ccd69a03ebb10811f04c0

Reply:
This seems like a worse UX though. Instead of providing the runtime config type (which should be fairly obvious), the user of the API has to supply this resolver type which requires the user to think more deeply about what that type actually is.

Reply:
🤔 doesn't the use already have instantiate the resolver in order to instantiate the factor?

Reply:
You aren't wrong.......

Reply:
Ok I changed to being generic over the resolver.